### ISSUE TITLE
OTI allows passing undefined for a non-undefined varargs param

### ISSUE THREAD

**Issue by shicks** [google/closure-compiler#2561]
[Example]

```js
var /** !Array<number> */ x = [];
x.push(undefined);
```

This should be an error, but OTI does not catch it.

[Example]: https://closure-compiler-debugger.appspot.com/#input0%3Dvar%2520%252F**%2520!Array%253Cnumber%253E%2520*%252F%2520x%2520%253D%2520%255B%255D%253B%250Ax.push(undefined)%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26MISSING_PROPERTIES%3D1%26TRANSPILE%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1

**Comment by brad4d** [comment]
Looks like I need to fix this as part of adding support for REST parameters.

**Comment by brad4d** [comment]
Testing a potential fix internally.
